ipatests/test_integration/test_server_del.py

```diff
@@ -300,3 +300,29 @@ def test_forced_removal_of_master(self):
             self.replicas[0], self.master.hostname,
             ignore_last_of_role=True
         )
+
+
+class TestServerDelReplicaRemoval(IntegrationTest):
+    """
+    Test the checks for Removal of Replica during server-del
+    as part for verification of bz1506188
```

Comment:
It should mainly refer to the upstream ticket #6572 not the bugzilla.  The bug was that dnsserver records were deleted after removal of a replica.  The test case description doesn't mention it, IMO it should mention it. E.g. something like: "Test that DNS server records and Server records are deleted during server-del." The case can be extended in a future with more checks.